File: JULY-2025/21-07-2025/p1957.cpp
class Solution {
public:
    string makeFancyString(string s) {
        int n = s.size();
        string ans = "";
        int count = 0;
        for(int i=0;i<n;i++){
            if(count==0){
                ans+=s[i];
                count++;
            }
            else if(s[i]==s[i-1]){
                if(count<2){
                    ans+=s[i];
                    count++;
                }
                else{
                    continue;
                }
            }
            else{
                ans+=s[i];
                count=1;
            }
        }
        return ans;
    }
};

File: JULY-2025/21-07-2025/p1957.swift
class Solution {
    func makeFancyString(_ s: String) -> String {
        let letters = Array(s)
        var ans = [Character]()
        for letter in letters {
            let count = ans.count
            if count < 2 || !(ans[count - 1] == ans[count - 2] && ans[count - 1] == letter) {
                ans.append(letter)
            }
        }
        return String(ans)        
    }
}